services/venice-controller/src/main/java/com/linkedin/venice/controller/HelixVeniceClusterResources.java

```diff
@@ -208,26 +206,6 @@ public HelixVeniceClusterResources(
     veniceAdminStats = new VeniceAdminStats(metricsRepository, "venice-admin-" + clusterName);
     this.storagePersonaRepository =
         new StoragePersonaRepository(clusterName, this.storeMetadataRepository, adapterSerializer, zkClient);
-    if (!config.isParent() && config.isSystemStoreHealthCheckEnabled()) {
```

Comment:
@gaojieliu Originally I thought we can make the change here, but considering the class is spinned up inside VeniceHelixAdmin, instead of VeniceParentHelixAdmin, repair task is unlikely to be created here, which requires parent admin

Reply:
An alternative is to leverage `ClusterLeaderInitializationRoutine` to register a callback from parent admin.
But I think the way this PR chooses is fine as well since `VeniceParentHelixAdmin#isLeaderControllerFor` is a cheap operation.